scripts/devops_tasks/python_snippet_updater.py

```diff
@@ -0,0 +1,94 @@
+import sys
+from pathlib import Path
+import re
+
+snippets = {}
+not_up_to_date = False
+
+target_snippet_sources = ["samples/*.py", "samples/*/*.py"]
+target_md_files = ["README.md"]
+
+def get_snippet(file):
+    with open(file, 'r') as f:
+        content = f.read()
+    pattern = "# \\[START[A-Z a-z0-9_]+\\][\\s\\S]+?# \\[END[A-Z a-z0-9_]+\\]"
```

Comment:
Just checking, given that this is a regex, nested snippets aren't supported with this. Totally fine with that yes?

Reply:
Right. We intentionally don't want to support nested snippets.